hack/build-flags.sh

```diff
@@ -24,6 +24,11 @@ function build_flags() {
     [[ -n "${commit}" ]] || (echo "error getting the current commit" && exit 1)
     version="v$(date +%Y%m%d)-local-${commit}"
   fi
+  # Knative component versions
+  local serving="`git ls-remote --tags --ref https://github.com/knative/serving.git | cut -d '-' -f2 | cut -d 'v' -f2 | sort -Vr | head -n 1`"
+  local kourier="`git ls-remote --tags --ref https://github.com/knative-sandbox/net-kourier.git | cut -d '-' -f2 | cut -d 'v' -f2 | sort -Vr | head -n 1`"
+  local eventing="`git ls-remote --tags --ref https://github.com/knative/eventing.git | cut -d '-' -f2 | cut -d 'v' -f2 | sort -Vr | head -n 1`"
```

Comment:
I think this would break with release branches - doesn't this always pick the latest?

Reply:
Good catch

Reply:
tested fix on 1.6 and 1.7 branches, works as expected